### ISSUE TITLE
Get rid of GLEW

### ISSUE THREAD

**Issue by pjurkas** [AcademySoftwareFoundation/OpenColorIO#547]
File src/lang-glsl/glsl.cpp uses GLEW. It does not call glewInit(). We have to call it from our main to prevent crashes. Would it be possible to get rid of GLEW?

This worked for me on Linux:

```diff
diff --git a/src/lang-glsl/glsl.cpp b/src/lang-glsl/glsl.cpp
index 44d1997..3cffde8 100644
--- a/src/lang-glsl/glsl.cpp
+++ b/src/lang-glsl/glsl.cpp
@@ -33,9 +33,9 @@ OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 #elif _WIN32
 #include <GL/glew.h>
 #else
-#include <GL/glew.h>
+#define GL_GLEXT_PROTOTYPES
 #include <GL/gl.h>
-#include <GL/glext.h>
+#include <GL/glu.h>
 #endif
```

**Comment by YashAgrawal0** [comment]
Is this issue still open?

**Comment by doug-walker** [comment]
@YashAgrawal0, yes this issue is still open.  @hodoulp, could answer other questions on this.

**Comment by hodoulp** [comment]
@YashAgrawal0 The defect is still open and relevant.

As mentioned by @doug-walker, the `EGL` option could also be a very interesting investigation/fix.

**Comment by MOFD308** [comment]
If I want to take "Restore Java bindings" as my Google Summer of Code project, can any members tell me where to start? Thanks!

**Comment by michdolan** [comment]
Hi @MOFD308 - thanks for your interest in our GSoC involvement. I've created a new issue for the Java bindings where we can discuss this:

https://github.com/AcademySoftwareFoundation/OpenColorIO/issues/950

**Comment by ChinYing-Li** [comment]
Hi, is this issue still open? OP's reason for wanting to get rid of GLEW is to prevent crashes when glewInit() isn't called. As an alternative, libepoxy requires no initialization code.
I can also investigate the possibility of "replacing GLX with EGL". Any suggestions?

**Comment by hodoulp** [comment]
Hi @ChinYing-Li,
The defect is still open which could be an interesting 'project' if you have a linux machine.

But it represents two works. The first one is get rid off GLEW for linux platform only (as explained by @pjurkas). The second is to improve the GPU unit test framework to run on headless machines and the use of EGL library could allow that. It should be two pull requests as the second work needs some investigations & validations.

**Comment by ChinYing-Li** [comment]
Hi @hodoulp ,
Thank you for explaining. I think I can start working on the second part, since @doug-walker mentioned that this might remove the need for GLEW, though I haven't investigated deep into this issue yet. What do you think?

**Comment by hodoulp** [comment]
Agree. That's part of the investigation around EGL.

**Comment by ChinYing-Li** [comment]
Since OpenColorIO need a OpenGL loader, shall we go for custom gl loader implementation? Or shall we replace glew with a loader library that does not require init()?

**Comment by doug-walker** [comment]
I believe this was fixed by PR #1047.  Closing this issue.  If this is still a problem for anyone, please let us know.
